pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -1210,6 +1210,7 @@ var _ = Describe("Test Work Generator Controller", func() {
 					return errors.IsNotFound(err)
```

Comment:
please add a case where the work is stuck in deleting for both integration tests and e2e tests.